app/js/components/tabs/tabcomponents/conceptComponent.jsx

```diff
@@ -1,35 +1,153 @@
 import React, {Component} from 'react';
 
 class ConceptComponent extends Component {
-    componentDidMount(){}
+    constructor(props) {
+        super(props);
+        this.state = {
+            conceptsResults: [],
+            searchResults: [],
+            verbose: false
+        };
+        this.allConcepts = [];
+        this.foundConcepts = [];
+        this.verboseSearchResult = [];
+        this.searchConcept = this.searchConcept.bind(this);
+        this.checkVerbose = this.checkVerbose.bind(this);
+        this.loadConcepts = this.loadConcepts.bind(this);
+    }
+    /**
+     * Make call to /concepts to fetch all concepts and load into allConcepts array
+     * @param {*} props 
```

Comment:
@dkayiwa just to add to what @andela-odaniel said, we're loading only the concept names into an array, not the entire concept data. This is so that we don't have to make a server-side call each time a user enters a single letter, we just loop through the already stored concept names.